core/string/ustring.cpp

```diff
@@ -2356,28 +2356,25 @@ bool String::is_numeric() const {
 }
 
 template <class C>
-static double built_in_strtod(const C *string, /* A decimal ASCII floating-point number,
-				 * optionally preceded by white space. Must
-				 * have form "-I.FE-X", where I is the integer
-				 * part of the mantissa, F is the fractional
-				 * part of the mantissa, and X is the
-				 * exponent. Either of the signs may be "+",
-				 * "-", or omitted. Either I or F may be
-				 * omitted, or both. The decimal point isn't
-				 * necessary unless F is present. The "E" may
-				 * actually be an "e". E and X may both be
-				 * omitted (but not just one). */
+static double built_in_strtod(const C *string, /* A decimal ASCII floating-point
+		 * number, optionally preceded by white space. Must have form "-I.FE-X",
+		 * where I is the integer part of the mantissa, F is the fractional part
+		 * of the mantissa, and X is the exponent. Either of the signs may be "+",
+		 * "-", or omitted. Either I or F may be omitted, or both. The decimal
+		 * point isn't necessary unless F is present. The "E" may actually be an
+		 * "e". E and X may both be omitted (but not just one). */
 		C **endPtr = nullptr) /* If non-nullptr, store terminating Cacter's
```

Comment:
Shouldn't we be using multiple `//` instead of `/* ... */`?

Reply:
That's my preference too, but here this reads like some thirdparty code that was copy-pasted from somewhere (hopefully public domain...?) and I didn't want to dig too deep down the rabbit hole refactoring it.

But since it's been there since GODOT IS OPEN SOURCE I guess it doesn't hurt to tweak it further to match our style yeah.

Reply:
Fixed.

Reply:
> this reads like some thirdparty code that was copy-pasted from somewhere (hopefully public domain...?) and I didn't want to dig too deep down the rabbit hole refactoring it.

Dug a bit and indeed it's some 1994 Sun Microsystems code found in Java, Tcl, Python, which we're not properly attributing. Maybe it's completely unneeded now and we could rely on some equivalent C/C++ standard lib function?

Reply:
I reverted my changes to the file for this PR, I think we should work on getting rid of this thirdparty code which we're not properly attributing. It's likely that it was added at a time where the C standard was not mature enough for a reliable cross-platform solution, but now we can use C11/C++11 so probably the standard `strtod` or `atof` might be good to use? https://www.cplusplus.com/reference/cstdlib/strtod/